### PR TITLE
planner: update the document about Plan-Cache

### PR DESCRIPTION
**Pull request by qw4990**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Comment by TomShawn** [comment]
@qw4990 这是 v5.3 Plan Cache GA 的文档吗？

**Review by Reminiscent — APPROVED** [review]

File: sql-prepare-plan-cache.md [hunk 1]
```diff
@@ -14,16 +14,20 @@ TiDB 优化器对这两类查询的处理是一样的：`Prepare` 时将参数
...
-- 查询包含子查询；
-- 查询包含不能被缓存的函数表达式，如 `current_user()`、`database()`、`last_insert_id()` 等；
-- 查询的 `Order By` 语句中包含 `?`；
-- 查询的 `Group By` 语句中包含 `?`；
-- 查询的 `Limit [Offset]` 语句中包含 `?`；
-- 查询包含的 `Window` 函数的 window frame 定义含有 `?`；
-- 查询引用了分区表；
+在当前版本中，当 `Prepare` 语句符合以下条件任何一条，查询或者计划不会被缓存：
+
+- 不属于 `Select`、`Update`、`Insert`、`Delete` 和集合操作 `Union`、`Intersect`、`Except` 的查询；
```
Comment (TomShawn) [inline sql-prepare-plan-cache.md:19]:
```suggestion
- `SELECT`、`UPDATE`、`INSERT`、`DELETE` 以外的 SQL 语句，以及包含集合操作 `Union`、`Intersect`、`Except` 的查询语句；
```

File: sql-prepare-plan-cache.md [hunk 2]
```diff
@@ -14,16 +14,20 @@ TiDB 优化器对这两类查询的处理是一样的：`Prepare` 时将参数
...
+- 会访问分区表、临时表或访问表中包含生成列的查询；
```
Comment (TomShawn) [inline sql-prepare-plan-cache.md:20]:
```suggestion
- 访问分区表、临时表或访问表中包含生成列的查询；
```

File: sql-prepare-plan-cache.md [hunk 3]
```diff
@@ -14,16 +14,20 @@ TiDB 优化器对这两类查询的处理是一样的：`Prepare` 时将参数
...
+- 包含子查询的查询，如 `select * from t where a > (select ...)`；
+- 包含 `ignore_plan_cache` `hint` 的查询，如 `select /*+ ignore_plan_cache() */ * from t`；
```
Comment (TomShawn) [inline sql-prepare-plan-cache.md:22]:
```suggestion
- 包含 `ignore_plan_cache` 这一 Hint 的查询，例如 `select /*+ ignore_plan_cache() */ * from t`；
```

File: sql-prepare-plan-cache.md [hunk 4]
```diff
@@ -14,16 +14,20 @@ TiDB 优化器对这两类查询的处理是一样的：`Prepare` 时将参数
...
+- 包含除 `?` 外其他变量（系统变量或用户自定义变量）的查询，如 `select * from t where a>? and b>@x`；
```
Comment (TomShawn) [inline sql-prepare-plan-cache.md:23]:
```suggestion
- 包含除 `?` 外其他变量（即系统变量或用户自定义变量）的查询，例如 `select * from t where a>? and b>@x`；
```

File: sql-prepare-plan-cache.md [hunk 5]
```diff
@@ -14,16 +14,20 @@ TiDB 优化器对这两类查询的处理是一样的：`Prepare` 时将参数
...
+- 包含无法被缓存函数的查询，目前不能被缓存的函数有：`database()`、`current_user`、`current_role`、`user`、`connection_id`、`last_insert_id`、`row_count`、`version`、`like`；
```
Comment (TomShawn) [inline sql-prepare-plan-cache.md:24]:
```suggestion
- 查询包含无法被缓存函数。目前不能被缓存的函数有：`database()`、`current_user`、`current_role`、`user`、`connection_id`、`last_insert_id`、`row_count`、`version`、`like`；
```

File: sql-prepare-plan-cache.md [hunk 6]
```diff
@@ -14,16 +14,20 @@ TiDB 优化器对这两类查询的处理是一样的：`Prepare` 时将参数
...
+- `?` 在 `Limit` 后的查询，如 `Limit ?` 或者 `Limit 10, ?`，此时 `?` 的具体值对查询影响较大，故不缓存；
+- `?` 直接在 `Order By` 后的查询，如 `Order By ?`，此时 `?` 表示根据 `Order By` 后第几列排序，对查询影响较大，故不缓存；如果是普通表达式，如 `Order By a+?` 则可以缓存；
```
Comment (TomShawn) [inline sql-prepare-plan-cache.md:26]:
```suggestion
- `?` 直接在 `Order By` 后的查询，如 `Order By ?`，此时 `?` 表示根据 `Order By` 后第几列排序，对查询影响较大，故不缓存；如果是普通表达式，如 `Order By a+?` 则会缓存；
```

File: sql-prepare-plan-cache.md [hunk 7]
```diff
@@ -14,16 +14,20 @@ TiDB 优化器对这两类查询的处理是一样的：`Prepare` 时将参数
...
+- `?` 直接在 `Group by` 后的查询，如 `Group By ?`，此时 `?` 表示根据 `Group By` 后第几列聚合，对查询影响较大，故不缓存；如果是普通表达式，如 `Group By a+?` 则可以缓存；
```
Comment (TomShawn) [inline sql-prepare-plan-cache.md:27]:
```suggestion
- `?` 紧跟在 `Group by` 后的查询，如 `Group By ?`，此时 `?` 表示根据 `Group By` 后第几列聚合，对查询影响较大，故不缓存；如果是普通表达式，如 `Group By a+?` 则会缓存；
```

File: sql-prepare-plan-cache.md [hunk 8]
```diff
@@ -14,16 +14,20 @@ TiDB 优化器对这两类查询的处理是一样的：`Prepare` 时将参数
...
+- `?` 出现在窗口函数 `Window Frame` 定义中的查询，如 `(partition by year order by sale rows ? preceding)`；如果 `?` 出现在窗口函数的其他位置，可以使用缓存；
```
Comment (TomShawn) [inline sql-prepare-plan-cache.md:28]:
```suggestion
- `?` 出现在窗口函数 `Window Frame` 定义中的查询，如 `(partition by year order by sale rows ? preceding)`；如果 `?` 出现在窗口函数的其他位置，可以会缓存；
```

File: sql-prepare-plan-cache.md [hunk 9]
```diff
@@ -39,8 +43,9 @@ key 中任何一项变动（如切换数据库，重命名 `Prepare` 语句，
 
 在通过合法性检测后，会根据当前最新参数值，对执行计划的扫描范围做相应调整，再用它执行获取数据。
 
-关于执行计划缓存和查询性能有两点值得注意：
+关于执行计划缓存和查询性能有几点值得注意：
 
+- 第一次执行 `Execute` 时（此时还没有缓存计划），会受到已有 `SQL Binding` 的影响；但是已经缓存的计划，不会受到新创建的 `SQL Binding` 的影响。同理，已经缓存的计划也不会受到统计信息更新、优化规则和表达式下推黑名单更新的影响。
```
Comment (TomShawn) [inline sql-prepare-plan-cache.md:48]:
```suggestion
- 第一次执行 `Execute` 时（此时还没有缓存计划），会受到已有 SQL Binding 的影响；但是已经缓存的计划，不会受到新创建的 SQL Binding 的影响。同理，已经缓存的计划也不会受到统计信息更新、优化规则和表达式下推黑名单更新的影响。
```

**Review by TomShawn — APPROVED** [review]
These are only minor comments on wording.

**Comment by TomShawn** [comment]
是否需要说明 Plan Cache 会在 v5.3 中 GA 的信息呢？

File: sql-prepare-plan-cache.md [hunk 10]
```diff
@@ -46,6 +46,7 @@ key 中任何一项变动（如切换数据库，重命名 `Prepare` 语句，
 关于执行计划缓存和查询性能有几点值得注意：
 
 - 第一次执行 `Execute` 时（此时还没有缓存计划），会受到已有 SQL Binding 的影响；但是已经缓存的计划，不会受到新创建的 SQL Binding 的影响。同理，已经缓存的计划也不会受到统计信息更新、优化规则和表达式下推黑名单更新的影响。
+- 升级或重启 TiDB 实例时，`Prepare` 信息可能被丢弃，如果一开始执行了 `prepare stmt ...`，后面只执行 `execute stmt ...`；这样当升级 TiDB 时，会返回 `Prepared Statement not found` 的错误，需要再次执行一次 `prepare stmt ...`。
```
Comment (Reminiscent) [inline sql-prepare-plan-cache.md:49]: `Prepare` 信息可能被丢弃 -> `Prepare` 信息会被丢弃
这样当升级 TiDB 时，会返回 `Prepared Statement not found` 的错误 ->  这样在不停机升级 TiDB 集群时，执行 `execute stmt ...` 可能会遇到 `Prepared Statement not found` 的错误

**Review by Reminiscent — APPROVED** [review]

**Review by Reminiscent — APPROVED** [review]

File: sql-prepare-plan-cache.md [hunk 11]
```diff
@@ -14,16 +14,20 @@ TiDB 优化器对这两类查询的处理是一样的：`Prepare` 时将参数
...
+- `SELECT`、`UPDATE`、`INSERT`、`DELETE`、`Union`、`Intersect`、`Except` 以外的 SQL 语句；
+- 访问分区表、临时表或访问表中包含生成列的查询；
+- 包含子查询的查询，如 `select * from t where a > (select ...)`；
+- 包含 `ignore_plan_cache` 这一 Hint 的查询，例如 `select /*+ ignore_plan_cache() */ * from t`；
+- 包含除 `?` 外其他变量（即系统变量或用户自定义变量）的查询，例如 `select * from t where a>? and b>@x`；
+- 查询包含无法被缓存函数。目前不能被缓存的函数有：`database()`、`current_user`、`current_role`、`user`、`connection_id`、`last_insert_id`、`row_count`、`version`、`like`；
+- `?` 在 `Limit` 后的查询，如 `Limit ?` 或者 `Limit 10, ?`，此时 `?` 的具体值对查询影响较大，故不缓存；
+- `?` 直接在 `Order By` 后的查询，如 `Order By ?`，此时 `?` 表示根据 `Order By` 后第几列排序，对查询影响较大，故不缓存；如果是普通表达式，如 `Order By a+?` 则会缓存；
+- `?` 紧跟在 `Group by` 后的查询，如 `Group By ?`，此时 `?` 表示根据 `Group By` 后第几列聚合，对查询影响较大，故不缓存；如果是普通表达式，如 `Group By a+?` 则会缓存；
+- `?` 出现在窗口函数 `Window Frame` 定义中的查询，如 `(partition by year order by sale rows ? preceding)`；如果 `?` 出现在窗口函数的其他位置，可以会缓存；
```
Comment (shichun-0415) [inline sql-prepare-plan-cache.md:28]:
```suggestion
- `?` 出现在窗口函数 `Window Frame` 定义中的查询，如 `(partition by year order by sale rows ? preceding)`；如果 `?` 出现在窗口函数的其他位置，则会缓存；
```

**Review by TomShawn — APPROVED** [review]
LGTM

**Pull request merged**